Reconnaissance demo tutorial, rework of #19

vmayoral opened the pull request.

vmayoral committed:
First iteration on reconnaissance demo

- Add aztarna (from source for development)
- Add configuration files for demo
- Add README.md instructions

vmayoral committed:
reconnaissance demo update

- Update Dockerfile
- Complete README.md
- Update dir naming

vmayoral committed:
Fix WORKDIR back and minor typo correction

vmayoral committed:
Remove reconnaissance-specific config files, minor fixes

vmayoral committed:
Update README.md

Co-Authored-By: Ruffin <roxfoxpox@gmail.com>

vmayoral committed:
Update README.md

Co-Authored-By: Ruffin <roxfoxpox@gmail.com>

vmayoral committed:
Update README.md

Co-Authored-By: Ruffin <roxfoxpox@gmail.com>

vmayoral committed:
Optimize cache with changes,

Attend https://github.com/ros-swg/turtlebot3_demo/pull/19/files/fb054e1478e74ec9156ff360d68eb6df4ec558fa#r337690173

vmayoral committed:
Further improve reconnaissance section

Address comments and reviews. Credit to @ruffsl

vmayoral committed:
Minor fix in Dockerfile for reconnaissance

vmayoral commented:
@mikaelarguedas could you please review this one?

mikaelarguedas commented on Dockerfile:
```diff
@@ -52,6 +52,21 @@ RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+RUN git clone https://github.com/aliasrobotics/aztarna && \
+      cd aztarna && python3 setup.py install
+# install tshark
+RUN apt-get -qq update && DEBIAN_FRONTEND=noninteractive apt-get -y install tshark
```
Nitpick: Could this be merged with the install layer just above?

vmayoral replied:
Done :)!

mikaelarguedas commented on README.md:
````diff
@@ -57,6 +57,46 @@ Feel free to poke around, open a new window and list or echo topics and services
+
+### Host/container insider attacker
+``` bash
+rocker rosswg/turtlebot3_demo:roscon19 "byobu -f configs/unsecure.conf attach"
````
This fails for me because --x11 is not passed as an argument. Adding --nvidia would be good for consistency with the rest of the tutorial

vmayoral replied:
Sent as well. Thanks!

mikaelarguedas reviewed (approved):
Looks good! I tried just the reconnaissance section and it works fine :+1:

Nitpick comments inline

vmayoral committed:
Merge install layers

vmayoral committed:
Add flags for consistency

vmayoral committed:
Silly mistake, fix

ruffsl commented on configs/common.conf:
```diff
@@ -47,6 +47,15 @@ send-keys 'ros2 security generate_artifacts -k keystore -p policies/tb3_gazebo_p
+send-keys 'aztarna -t ros2'
+
+split-window -h
+send-keys 'tshark -i eth0 -f "udp" -Y "rtps"' Enter
```
I don't think we want to run the command by default for every demo. Perhaps remove the Enter, and remind the reader to press enter in the command window when starting the tutorial. Or add this window using a separate root level config, like we have for secure and unsecure.

vmayoral replied:
That's fair. Let me remove it then.

vmayoral replied:
9b76724e1586e2a0fc28d938bb7da7fbb6dcb8f0

vmayoral committed:
Remove Enter from tshark command

Attend https://github.com/ros-swg/turtlebot3_demo/pull/31#discussion_r339304585
